owa/app/js/components/manage_patient_identifier_sources/newPatientIdentifierSources.jsx

```diff
@@ -0,0 +1,135 @@
+import React, {Component} from 'react';
+import { Table, Input, Button, Form, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import LocalIdentifierGenerator from './new_patient_identifier_sources/localIdentifierGenerator'
+import LocalPoolIdentifiers from './new_patient_identifier_sources/localPoolIdentifiers'
+import RemoteIdentifierSource from './new_patient_identifier_sources/remoteIdentifierSource'
+
+export default class NewPatientIdentifierSources extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      modalContent: 'Please select a Source Type',
+      title: 'New Patient Identifier Source',
+      identifierType: 'Please select an identifier Type'
+    };
+
+    this.toggle = this.toggle.bind(this);
+    this.getSource = this.getSource.bind(this);
+    this.getIdentifier = this.getIdentifier.bind(this);
+  }
+
+  // display the modal
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
+    });
+  }
+
+  //get the identifier type
+  getIdentifier(){
+    let options = document.getElementById("identifierType").options
+    let selectedIndex = document.getElementById("identifierType").selectedIndex
+    this.setState({
+      identifierType: options[selectedIndex].text
+    }, () => {
+      this.getSource()
+    });
+
+  }
+
+  //get the source type
+  getSource() {
+    switch(document.getElementById("sourceType").value){
+      case "org.openmrs.module.idgen.SequentialIdentifierGenerator":
+        this.setState({
```

Comment:
Reach out to Bamidele, I sent him a guide on how to use functional setState instead of setting state with an object

Reply:
I have implemented functional setState.